volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,646 @@
+                product_name=product,
+                company=publisher,
+            )
+
+    @classmethod
+    def parse_inventory_app_file_key(
+        cls, inv_app_file_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[Tuple[NullableString, _AmcacheEntry]]:
+        """Parses executable file entries from the `Root\\InventoryApplicationFile` registry key.
+
+        :param inv_app_file_key: The `Root\\InventoryApplicationFile` registry key.
+        :return: An iterator of tuples, where the first member is the program ID string for correlating
+        with it's parent `InventoryApplication` program entry, and the second member is the `Amcache` entry.
+        """
+
+        valName = Win10InvAppFileValName
+
+        for file_key in inv_app_file_key.get_subkeys():
+
+            vollog.debug(
+                f"Parsing Win10 InventoryApplicationFile key {file_key.get_name()}"
+            )
+
+            values = {
+                str(value.get_name()): value
+                for value in file_key.get_values()
+                if value.get_name() in [key.value for key in valName]
+            }
+
+            last_mod = conversion.wintime_to_datetime(file_key.LastWriteTime.QuadPart)
+            path = _get_string_value(values, valName.LowerCaseLongPath.value)
+            linkdate = _get_datetime_str_value(values, valName.LinkDate.value)
+            sha1_hash = _get_string_value(values, valName.FileId.value)
+            publisher = _get_string_value(values, valName.Publisher.value)
+            prod_name = _get_string_value(values, valName.ProductName.value)
+            prod_ver = _get_string_value(values, valName.ProductVersion.value)
+            program_id = _get_string_value(values, valName.ProgramID.value)
+
+            if isinstance(prod_ver, str):
+                if isinstance(prod_name, str):
+                    prod_name = f"{prod_name} {prod_ver}"
+                else:
+                    prod_name = f"UNKNOWN {prod_ver}"
+
+            yield program_id, _AmcacheEntry(
+                AmcacheEntryType.File.name,
+                path=path,
+                company=publisher,
+                last_modify_time=last_mod,
+                compile_time=linkdate,
+                sha1_hash=(
+                    sha1_hash.lstrip("0000")
+                    if isinstance(sha1_hash, str)
+                    else sha1_hash
+                ),
+                product_name=prod_name,
+            )
+
+    @classmethod
+    def parse_driver_binary_key(
+        cls, driver_binary_key: reg_extensions.CM_KEY_NODE
+    ) -> Iterator[_AmcacheEntry]:
+        """Parses information about installed drivers from the `Root\\InventoryDriverBinary` registry key.
+
+        :param driver_binary_key: The `Root\\InventoryDriverBinary` registry key
+        :return: An iterator of `AmcacheEntry`s
+        """
+        for binary_key in driver_binary_key.get_subkeys():
+
+            valName = Win10DriverBinaryValName
```

Comment:
This doesn't need to be reset each time round the loop, feels like it could be set once before the loop starts...